sha1_mb/sha1_mb_x4_sse.asm

```diff
@@ -77,17 +77,18 @@ default rel
 ;;
 ;; Magic functions defined in FIPS 180-1
 ;;
-; macro MAGIC_F0 F,B,C,D,T   ;; F = (D ^ (B & (C ^ D)))
+; macro MAGIC_F0 F,B,C,D,T   ;; F = ((B & C) | ((~ B) & D))
 %macro MAGIC_F0 5
 %define %%regF %1
 %define %%regB %2
 %define %%regC %3
 %define %%regD %4
 %define %%regT %5
-    movdqa  %%regF,%%regC
-    pxor  %%regF,%%regD
-    pand  %%regF,%%regB
-    pxor  %%regF,%%regD
+    movdqa %%regF, %%regB
```

Comment:
Yes, sorry about that, i ORed the register with itself, guess it didn't show up when i tested it on my machine because i've a Skylake CPU and the test pics up the AVX2 codepath, let me know if it works now :)

Reply:
That fixed it but I'm measuring the SSE version is about 4% slower with the change.  We are probably instruction bound in this case.

Reply:
Interesting, so it's probabily safe to keep the old macrp for the SSE version?
I'm curious, on what CPU was the test run? The movs reg,reg should be eliminated in the rename stage on modern CPU but i've noticed that it does not happen as often as i thought it would and unfortunately Intel's Optimization Manual doesn't explain the new microarchitectures in any kind of depth :(

Reply:
Agreed.  Let's keep the old SSE macro.  I tested on an older IA prior to AVX and also an Atom and they came in at -1% and -4% respectively.